contrib/odata-hierarchies-2-sql/sqlite3/traverse-example-section-3.3.sql

```diff
@@ -0,0 +1,98 @@
+.read "create-schema.sql"
+
+-- logic for a depth first search traversal of a hierarchy
+with recursive
+  -- traversal needs a stable sibling order; can be rowid if selecting from hierarchy table, or row_number() if preceding transformation
+  salesorg_w_sibling(ID,Superordinate,Name,SiblingIdx) as (select *,rowid from SalesOrganization),
+  parent_of(node,parent_node) as (select id, Superordinate from salesorg_w_sibling),
+  dfs_traversal_of(node, path) as (
+    select id, '0001' from salesorg_w_sibling -- start with root node
+     where Superordinate is null 
+     union
+    select parent_of.node,
+           path || '.' || substr('000' || cast(SiblingIdx as string),-4) -- build lexicographical sequence for each node
```

Comment:
I suggest to refactor and do the conversion from rowId to a four digit number with leading zeros in the parent_of tab table expression. 
That way 
a) the expression here becomes simpler
b) if the number of digits has to be increased the change is in one place (in contrast to here and in the base case of the recursive CTE)